sdk/monitor/azure-monitor-query/src/test/java/com/azure/monitor/query/MetricsBatchQueryClientTest.java

```diff
@@ -3,34 +3,81 @@
 
 package com.azure.monitor.query;
 
+import com.azure.core.exception.HttpResponseException;
 import com.azure.core.http.policy.HttpLogDetailLevel;
 import com.azure.core.http.policy.HttpLogOptions;
-import com.azure.core.test.annotation.RecordWithoutRequestBody;
 import com.azure.core.util.Configuration;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.data.appconfiguration.ConfigurationClient;
+import com.azure.data.appconfiguration.ConfigurationClientBuilder;
+import com.azure.monitor.query.models.MetricResult;
 import com.azure.monitor.query.models.MetricsBatchResult;
+import com.azure.monitor.query.models.MetricsQueryOptions;
+import com.azure.monitor.query.models.QueryTimeInterval;
 import org.junit.jupiter.api.Test;
 
+import java.time.Duration;
+import java.time.OffsetDateTime;
 import java.util.Arrays;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 /**
  * Unit tests for {@link MetricsBatchQueryClient}.
  */
 public class MetricsBatchQueryClientTest extends MetricsBatchQueryTestBase {
 
     @Test
-    @RecordWithoutRequestBody
     public void testMetricsBatchQuery() {
         MetricsBatchQueryClient metricsBatchQueryClient = clientBuilder
             .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
             .buildClient();
-        String resourceId = Configuration.getGlobalConfiguration().get("AZURE_MONITOR_METRICS_RESOURCE_URI", FAKE_RESOURCE_ID);
+        String resourceId = Configuration.getGlobalConfiguration().get("AZURE_MONITOR_METRICS_RESOURCE_URI_2", FAKE_RESOURCE_ID);
         resourceId = resourceId.substring(resourceId.indexOf("/subscriptions"));
 
-        MetricsBatchResult metricsQueryResults = metricsBatchQueryClient.queryBatch(
-            Arrays.asList(resourceId),
-            Arrays.asList("Successful Requests"), " Microsoft.Eventhub/Namespaces");
+        ConfigurationClient configClient = new ConfigurationClientBuilder()
+            .connectionString(Configuration.getGlobalConfiguration().get("AZURE_APPCONFIG_CONNECTION_STRING"))
```

Comment:
I think the tests are failing because we need a fill-in value here for playback.